tests/object/tree/read.c

```diff
@@ -73,3 +73,71 @@ void test_object_tree_read__two(void)
 	git_object_free(obj);
 	git_tree_free(tree);
 }
+
+#define BIGFILE "bigfile"
+#define BIGFILE_SIZE (off_t)4 * 1024 * 1024 * 1024 /* 4 GiB */
+
+void test_object_tree_read__largefile(void)
+{
+	git_reference *ref;
+	git_commit *commit;
+	git_tree *tree;
+	git_oid oid;
+	const git_tree_entry *entry;
+	git_object *object;
+	git_buf file = GIT_BUF_INIT;
+	int fd;
+#ifdef __APPLE__
+	fstore_t prealloc;
+#endif
+	git_index *idx;
+
+#ifdef GIT_WIN32
+	cl_skip();
```

Comment:
This concept [does exist in win32](https://blogs.msdn.microsoft.com/oldnewthing/20150710-00/?p=45171).  It might be nice to have a `p_fallocate` that abstracts the macOS / win32 / POSIX functionality.

Reply:
(If you want me to do the win32 bits, I'm happy to.)

Reply:
Yeah, it's just that I wasn't sure adding something to our POSIX layer only for a testcase was a nice use of library size (though maybe it can be deadcode-ed). If you're fine with this, I'd definitely prefer going that route.

Reply:
I ended up doing just that, and there's a WIN32 stub (for now 😉).